src/node_env_var.cc

```diff
@@ -389,13 +390,33 @@ static void EnvEnumerator(const PropertyCallbackInfo<Array>& info) {
       env->env_vars()->Enumerate(env->isolate()));
 }
 
+static void EnvDefiner(Local<Name> property,
+                       const PropertyDescriptor& desc,
+                       const PropertyCallbackInfo<Value>& info) {
+  Environment* env = Environment::GetCurrent(info);
+  if (desc.has_get() || desc.has_set() || desc.has_configurable() ||
+      desc.has_enumerable() || desc.has_writable()) {
+    env->ThrowError("Cannot specify data or accessor descriptors"
```

Comment:
This should be a `TypeError`.

Also, theoretically, for errors thrown C++ land that are not formatted by V8, we also need to bump major if they don't have a code, so preferably this should use the macros in `node_errors.h` or otherwise we'd need to bump major if someone ever wants to touch this message.

Reply:
I've changed to `TypeError`. And how to use `node_errors.h`? Which `V` should this need or just add a new `V`

Reply:
> Which V should this need or just add a new V

Yes, please add a new `V` with the message and the type (I think `ERR_INVALID_DEFINE` would be enough), then you can just do `THROW_ERR_INVALID_DEFINE` here. Please document the error code in `doc/api/errors.md` as ell.

Reply:
perfect

Reply:
resolved